src/base/bittorrent/filterparserthread.cpp

```diff
@@ -133,11 +196,8 @@ int FilterParserThread::parseDATFilterFile()
         return ruleCount;
     }
 
-    std::vector<char> buffer(BUFFER_SIZE, 0); // seems a bit faster than QVector
-    qint64 bytesRead = 0;
-    int offset = 0;
-    int start = 0;
-    int endOfLine = -1;
+    std::array<char, BUFFER_SIZE> buffer = {}; // seems a bit faster than QVector
```

Comment:
2MB on stack!? That is just too big, most platforms only have ~1MB by default.

Reply:
Hmm, could it be this the reason why mingw outright crashes? I'll investigate.
I suppose I should change it back to std::vector.